### DISCUSSION TITLE
XunitTestFrameworkExecutor has hidden dependency to XunitTestFrameworkDiscoverer

### DISCUSSION THREAD

**Discussion by dainius-bekeris** [xunit/xunit#2158] [General]
Found this hard way: one who wants custom discoverer shall also create specialized executor and override CreateDiscoverer() method. While it's possible to solve it without changes in framework, I think it would be better to change design: make dependency explicit, update XunitTestFramework implementation so that it's enough to override only discoverer creation.

Do you think it's possible to include such change into framework? I can do all needed changes.

**Comment by bradwilson** [comment]
Transferred to issue https://github.com/xunit/xunit/issues/2163
